tests/platform/test_reboot.py

```diff
@@ -66,6 +66,16 @@
 }
 
 
+@pytest.fixture(scope="module", autouse=True)
+def teardown_module(duthost, conn_graph_facts):
+    yield
+
+    logging.info("Tearing down: to make sure all the critical services, interfaces and transceivers are good")
+    interfaces = conn_graph_facts["device_conn"]
+    check_critical_services(duthost)
+    check_further_interfaces_and_services(duthost, interfaces)
```

Comment:
WHy not having single function for basic_sanity_check that 
1. verify critical services
2. verify services
3. verify interfaces
4. will come later

you will be able to extend it with minimal change in the basic_sanity_check function in the future.
also, i suggest that each failure will raise a detailed info of what was verified and why the validation failed

Reply:
Hi @liat-grozovik,
There is a common wrapper regarding all the condition checkings:
`wait_until(longest_time, delay, check_function)`
In fact, it's a retry logic, which means the check_function will be repeatedly called many times until success or maximum time reached. The check_function returns only a boolean result. But what is checked and what fails the check is known by check_function only. 
So the only way to raise the detailed info is to log the info when it fails from the check_function.
How do you think?

Reply:
Provide a single function for all kinds of sanity check.
Promote level of logs that report the state of services or interfaces from debug to info. This kind of logs won’t be too much but contain import information, so it’s reasonable to promote the log level. By doing so it’s able to provide clear information to user especially when the test fails.